flow/project.py

```diff
@@ -1809,7 +1827,89 @@ def _fetch_scheduler_status(self, jobs=None, file=None, ignore_errors=False):
         else:
             logger.info("Updated job status cache.")
 
-    def _fetch_status(self, jobs, err, ignore_errors, status_parallelization='thread'):
+    def _get_group_status(self, group_name, ignore_errors=False, cached_status=None):
+        "Return a dict with detailed information about the status of jobs per group."
+        result = dict()
+        group = self._groups[group_name]
+        # The group associated with the group_name contains only a
+        # single operation having its name equals group_name.
+        # Hence set the operation_name to group_name
+        result['operation_name'] = group_name
+        status_dict = dict()
+        errors = dict()
+        aggregates = self._get_aggregates(group.name)
+        for aggregate in tqdm(aggregates,
+                              desc=f"Collecting aggregate status info for operation {group.name}",
+                              leave=False):
+            errors.setdefault(get_aggregate_id(aggregate), '')
+            try:
+                job_op_id = group._generate_id(aggregate)
+                completed = group._complete(aggregate)
+                eligible = False if completed else group._eligible(aggregate)
+                status_dict[get_aggregate_id(aggregate)] = {
+                        'scheduler_status': cached_status.get(job_op_id, JobStatus.unknown),
+                        'eligible': eligible,
+                        'completed': completed,
+                        }
+            except Exception as error:
+                msg = "Error while getting operations status for aggregate " \
+                      "'{}': '{}'.".format(get_aggregate_id(aggregate), error)
+                logger.debug(msg)
+                status_dict[get_aggregate_id(aggregate)] = {
+                        'scheduler_status': JobStatus.unknown,
+                        'eligible': False,
+                        'completed': False,
+                        }
+                if ignore_errors:
+                    errors[get_aggregate_id(aggregate)] += str(error) + '\n'
+                else:
+                    raise
+
+        result['job_status_details'] = status_dict
+        result['_operation_error_per_job'] = errors
+        return result
+
+    def _get_job_labels(self, job, ignore_errors=False):
+        "Return a dict with information about the labels of a job."
+        result = dict()
+        result['job_id'] = str(job)
+        try:
+            result['labels'] = sorted(set(self.labels(job)))
+            result['_labels_error'] = None
+        except Exception as error:
+            logger.debug("Error while determining labels for job '{}': '{}'.".format(job, error))
+            if ignore_errors:
+                result['labels'] = list()
+                result['_labels_error'] = str(error)
+            else:
+                raise
+        return result
+
+    def _fetch_status(self, aggregates, distinct_jobs, err, ignore_errors,
```

Comment:
@kidrahahjo Can you offer your insight on this?

Reply:
Since every job is acting as an aggregate internally, I believe that the change was necessary for this PR to be consistent with the code. 

`get_job_status` returns the status wrt only a single job/aggregate. Considering the fact that we only use aggregates for fetching per operation status and job for fetching job labels, we may continue facing the problem while parallelization as I am unable to figure out a way which lets us do both at the same time. And this is the main reason why there's a ton of extra logic being used to construct two separate objects in parallel and then combine them.

Reply:
Punt to #362